app/models/usage_update.rb

```diff
@@ -0,0 +1,213 @@
+class UsageUpdate < Base
+  LICENSE = "https://creativecommons.org/publicdomain/zero/1.0/"
+
+  def self.import_by_month(options={})
+    from_date = (options[:from_date].present? ? Date.parse(options[:from_date]) : Date.current).beginning_of_month
+    until_date = (options[:until_date].present? ? Date.parse(options[:until_date]) : Date.current).end_of_month
+
+    # get first day of every month between from_date and until_date
+    (from_date..until_date).select {|d| d.day == 1}.each do |m|
+      UsageUpdateImportByMonthJob.perform_later(from_date: m.strftime("%F"), until_date: m.end_of_month.strftime("%F"))
+    end
+
+    "Queued import for Report updated from #{from_date.strftime("%F")} until #{until_date.strftime("%F")}."
+  end
+
+  def self.import(options={})
+    # from_date = options[:from_date].present? ? Date.parse(options[:from_date]) : Date.current - 1.day
+    # until_date = options[:until_date].present? ? Date.parse(options[:until_date]) : Date.current
+
+    usage_update = UsageUpdate.new
+    # usage_update.queue_jobs(usage_update.unfreeze(from_date: from_date.strftime("%F"), until_date: until_date.strftime("%F")))
+    usage_update.queue_jobs 
```

Comment:
Does this work? No date range given, what is `import_by_month` doing?

Reply:
At the moment Im not using `import_by_month` for anything. I took the crossref file as a template, that's why is there. 

You need to execute the rake task. And probably this is related to my question from earlier. How are we triggering this? Back then I think the idea was:

> As I said before, I want to use SQS to push things into levriero, not an API. Pull for backfilling and some agents, push via SQS if possible

So I did something similar to the other agents are doing. As far as I can see crossref agent goes to XrefEventData, the related_identifier goes to SORL. So the usageUpdate agent goes to SQS (queue: stage_usage). That's where SaSHIMI is pushing every new report and new update.